psutil_agent/tests/test_loop.py

```diff
@@ -0,0 +1,65 @@
+import sched
+import time
+
+import mock
+import testtools
+
+from psutil_agent.loop import collect_metrics, create_scheduled_fun
+
+
+class TestCollectMetrics(testtools.TestCase):
+    def test_no_method_and_no_interval(self):
+        with mock.patch('psutil_agent.loop.create_scheduled_fun') as m:
+            collect_metrics({}, [{}])
+            m.assert_not_called()
+
+    def test_no_interval(self):
+        with mock.patch('psutil_agent.loop.create_scheduled_fun') as m:
+            collect_metrics({}, [{'method': 'some_method'}])
+            m.assert_not_called()
+
+    def test_no_method(self):
+        with mock.patch('psutil_agent.loop.create_scheduled_fun') as m:
+            collect_metrics({}, [{'interval': 2}])
+            m.assert_not_called()
+
+    def test_both_method_and_interval_present(self):
+        with mock.patch('psutil_agent.loop.create_scheduled_fun') as m:
+            collect_metrics({}, [{'interval': 2, 'method': 'some method'}])
+            self.assertEqual(True, m.called,
+                             "create_scheduled_fun() wasn't called.")
+
+
+class TestCreateScheduledFun(testtools.TestCase):
+    scheduler = None
```

Comment:
maybe im missing something, but is this really needed in this scope? isnt this just an instance variable?